woocommerce/woocommerce-gutenberg-products-block
# Fix JS-rendered blocks inside Empty Cart

Fixes #2836.

This PR refactors how `renderFrontend` works so blocks that might contain JS-rendered inner blocks can trigger a `wc-blocks_render_blocks_frontend` JS event to render the inner blocks.

### Screenshots

![Peek 2020-08-03 20-10](https://user-images.githubusercontent.com/3616980/89213503-7b2c5d00-d5c5-11ea-9dce-7129102267d6.gif)

### How to test the changes in this Pull Request:

1. Add the Cart block and switch to the Empty Cart edit mode.
2. Replace the Newest products block with the All Products block. Feel free to add other JS-rendered blocks like a filter or a Reviews block.
3. View the page in the frontend.
4. Verify all blocks render correctly when the cart is empty. Try adding a product to the cart (so it switches to the Full Cart view) and removing it (so it switches back to the Empty Cart view). Verify the blocks still render correctly.

### Changelog

> Fixed an issue that was making some blocks not to render correctly in the Empty cart template.
